service/orchestrator/toe.go

```diff
@@ -109,14 +168,23 @@ func (svc *Service) GetTargetOfEvaluation(_ context.Context, req *orchestrator.G
 
 // ListTargetsOfEvaluation implements method for getting a TargetOfEvaluation
 func (svc *Service) ListTargetsOfEvaluation(_ context.Context, req *orchestrator.ListTargetsOfEvaluationRequest) (res *orchestrator.ListTargetsOfEvaluationResponse, err error) {
+	var conds = []any{gorm.WithPreload("ControlsInScope")}
+
 	// Validate request
 	err = service.ValidateRequest(req)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO: authz
+	if req.GetCloudServiceId() != "" {
+		conds = append(conds, "cloud_service_id = ?", req.CloudServiceId)
+	}
+
+	// TODO: filter according to catalog (and acually make it mutually exclusive)
```

Comment:
Can we achieve this somehow with the validation framework?

Reply:
Do we have to check that really? Either the binding `/v1/orchestrator/cloud_services/{cloud_service_id}/toes` with the cloud_service_id or `/v1/orchestrator/catalogs/{catalog_id}/toes` with the catalog_id is used. I would check if one of both is set and define the conds accordingly.

Reply:
That's true, and should probably be sufficient. I just wanted to make sure that an `invalid-argument` error is created if someone is issuing the gRPC call directly with neither the cloud service nor the catalog.

Reply:
Maybe it even makes sense to also have something like `/v1/orchestrator/toes` which displays all toes (currently visible to the session), mainly for debugging reasons, etc.

Reply:
We have the `/v1/orchestrator/toes` already: 
 ```
// Lists all Targets of Evaluation
  rpc ListTargetsOfEvaluation(ListTargetsOfEvaluationRequest)
      returns (ListTargetsOfEvaluationResponse) {
    option (google.api.http) = {
      get : "/v1/orchestrator/toes"
      additional_bindings {
        get : "/v1/orchestrator/cloud_services/{cloud_service_id}/toes"
      }
      additional_bindings {
        get : "/v1/orchestrator/catalogs/{catalog_id}/toes"
      }
    };
  }
```

I am not sure what you exactly mean.

Reply:
I've added the auth check to `ListTargetsOfEvaluation` and `CreateTargetOfEvaluationRequest`.

Reply:
Ok, I was getting confused apparently